website/docs/r/network_pool.html.markdown

```diff
@@ -0,0 +1,245 @@
+modify, and delete networks pools attached to a VCD.
+
+Supported in provider *v3.11+*
+
+~> Only `System Administrator` can create this resource.
+
+## Example Usage 1 - Type "GENEVE" (NSX-T)
+
+```hcl
+data "vcd_nsxt_manager" "mgr" {
+  name = "mymanager"
+}
+
+resource "vcd_network_pool" "npool" {
+  name                = "new-network-pool"
+  description         = "New network pool"
+  network_provider_id = data.vcd_nsxt_manager.mgr.id
+  type                = "GENEVE"
+
+  backing {
+    transport_zone {
+      name = "nsx-overlay-transportzone"
+    }
+  }
+}
+```
+
+## Example Usage 2 - Type "VLAN"
+
+```hcl
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+resource "vcd_network_pool" "npool" {
+  name                = "my-vlan-network-pool"
+  description         = "New VLAN network pool"
+  network_provider_id = data.vcd_vcenter.vc1.id
+  type                = "VLAN"
+
+  backing {
+    distributed_switch {
+      name = "NsxTDVS"
+    }
+    range_id {
+      start_id = 101
+      end_id   = 200
+    }
+  }
+}
+```
+## Example Usage 3 - Type "PORTGROUP_BACKED"
+
+```hcl
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+resource "vcd_network_pool" "npool" {
+  name                = "my-pg-network-pool"
+  description         = "New Port Group network pool"
+  network_provider_id = data.vcd_vcenter.vc1.id
+  type                = "PORTGROUP_BACKED"
+
+  backing {
+    port_group {
+      name = "TestbedPG"
+    }
+  }
+}
+```
+
+## Example Usage 4 Retrieving backing elements
+
+The elements needed as backing for a network pool can be retrieved using [`vcd_resource_list`](/providers/vmware/vcd/latest/docs/data_sources/resource_list), as in the example below
+
+```hcl
+data "vcd_nsxt_manager" "mgr" {
+  name = "nsxManager1"
+}
+
+data "vcd_vcenter" "vc1" {
+  name = "vc1"
+}
+
+data "vcd_resource_list" "tz" {
+  name          = "tz"
+  resource_type = "vcd_nsxt_transport_zone"
+  parent        = data.vcd_nsxt_manager.mgr.name
+}
+
+data "vcd_resource_list" "pg" {
+  name          = "pg"
+  resource_type = "vcd_importable_port_group"
+  parent        = data.vcd_vcenter.vc1.name
+}
+
+data "vcd_resource_list" "ds" {
+  name          = "ds"
+  resource_type = "vcd_distributed_switch"
+  parent        = data.vcd_vcenter.vc1.name
+}
+
+output "tzs" {
+  value = data.vcd_resource_list.tz.list
+}
+
+output "pgs" {
+  value = data.vcd_resource_list.pg.list
+}
+
+output "ds" {
+  value = data.vcd_resource_list.ds.list
+}
+```
+
+-> Note: the lists provided as `vcd_resource_list` output are volatile: they only exist for items that have not been used
+in a network pool. Once they have been assigned, they cease to be shown. As such, it is not a good idea to use
+`vcd_resource_list` as direct source for one or more network pools: at the first `plan`, terraform would propose
+to remove the network pool, as the element is not shown in the list anymore.
+
+## Example Usage 5 - nameless backing retrieval
```

Comment:
I'd like to clarify this section name. Suggestion:

```suggestion
## Example Usage 5 - automatic backing components selection
```

Reply:
Changed